src/pages/en/guides/environment-variables.md

```diff
@@ -16,15 +18,24 @@ Just create a `.env` file in the project directory and add some variables to it.
 
 ```bash
 # .env
+DB_PASSWORD="foobar"
 PUBLIC_POKEAPI="https://pokeapi.co/api/v2"
 ```
 
 ## Getting environment variables
 
 Instead of using `process.env`, with Vite you use `import.meta.env`, which uses the `import.meta` feature added in ES2020 (don't worry about browser support though, Vite replaces all `import.meta.env` mentions with static values). For example, to get the `PUBLIC_POKEAPI` environment variable, you could use `import.meta.env.PUBLIC_POKEAPI`.
 
-```js
-fetch(`${import.meta.env.PUBLIC_POKEAPI}/pokemon/squirtle`);
+```astro
+---
+// `.astro` only runs on the server, private variables work!
+const data = await db(import.meta.env.DB_PASSWORD);
+---
+
+<script type="module">
+  // scripts run on the client, only PUBLIC_ variables work!
+  const squirtle = await fetch(`${import.meta.env.PUBLIC_POKEAPI}/pokemon/squirtle`);
```

Comment:
did you test that this works? I thought we only processed scripts with `hoist` attribute. You may want to show a .jsx component instead.